src/middleware/TestingHandler.ts

```diff
@@ -0,0 +1,265 @@
+/**
+ * -------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.
+ * See License in the project root for license information.
+ * -------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @module TestingHandler
+ */
+
+import { Context } from "../IContext";
+
+import { Middleware } from "./IMiddleware";
+import { MiddlewareControl } from "./MiddlewareControl";
+import { generateUUID } from "./MiddlewareUtil";
+import { httpStatusCode, methodStatusCode } from "./options/TestingHandlerData";
+import { TestingHandlerOptions } from "./options/TestingHandlerOptions";
+import { TestingStrategy } from "./options/TestingStrategy";
+
+export class TestingHandler implements Middleware {
+	/**
+	 * @private
+	 * A member holding options to customize the handler behavior
+	 */
+
+	private options: TestingHandlerOptions;
+
+	public statusCode: number;
+
+	public statusMessage: string;
+
+	// the mode of testing handler
+	public testingStrategy: TestingStrategy;
+
+	// container for the manual map that has been written by the client
+	public manualMap: Map<string, Map<string, number>>;
+
+	// dummy url that we are using
+	public redirectURL: string = "https://dummylocation.microsoft.com";
```

Comment:
we should not rely on the hardcoded url. it is possible that user can set his base url as the one you provided. so we need to randomize this in order to reduce the possibility of the collision.